app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,67 @@
+class FoodsController < ApplicationController
+  before_action :set_food, only: %i[show edit update destroy]
+  # GET /foods or /foods.json
+  def index
+    @foods = current_user.foods
+  end
+
+  # GET /foods/1 or /foods/1.json
+  def show; end
+
+  # GET /foods/new
+  def new
+    @food = Food.new
+    respond_to do |format|
+      format.html { render :new, locals: { food: @food } }
+    end
+  end
+
+  # POST /foods or /foods.json
+  def create
+    @food = current_user.foods.build(food_params)
+    respond_to do |format|
+      if @food.save
+        format.html { redirect_to foods_path(@food), notice: 'Food was successfully created.' }
+        format.json { render :show, status: :created, location: @food }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @food.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /foods/1 or /foods/1.json
+  def update
+    respond_to do |format|
+      if @food.update(food_params)
+        format.html { redirect_to food_url(@food), notice: 'Food was successfully updated.' }
+        format.json { render :show, status: :ok, location: @food }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @food.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /foods/1 or /foods/1.json
+  def destroy
```

Comment:
- [ ] Nice work done on your project, kindly check when I click the delete button on the food page it throws the error below.
![error-food](https://user-images.githubusercontent.com/48371156/229289712-5042a7dd-0f9a-4913-8389-f76ac260cd16.png)